### ISSUE TITLE
ffprobe throws error for portrait videos on Windows

### ISSUE THREAD

**Issue by snapptop** [fluent-ffmpeg/node-fluent-ffmpeg#404]
```
var def = deferred();
ffmpeg.ffprobe(zpath, function (err, metadata) {
     if (err) return def.promise.emit('error', err);
     def.promise.emit('done', metadata);
});
return def.promise;
```
This works for landscape videos but fails for portrait videos.
This also breaks 'screenshots' method as it uses ffprobe.

Node Verison:  "fluent-ffmpeg": "^2.0.1",

FFMPEG Version:
This is a FFmpeg Win64 static build by Kyle Schwarz.
Zeranoe's FFmpeg Builds Home Page: <http://ffmpeg.zeranoe.com/builds/>
FFmpeg version: 2015-05-21 git-0b9d636

Is this a Windows only issue?

**Comment by njoyard** [comment]
Could you please give the error stack trace and ffprobe command line output for the failing video ?

**Comment by nickwhiteley** [comment]
I have this problem too.  A stack trace is included below.  data.format is undefined and data.streams is populated on line 152 of ffprobe.js and it's the undefined data.format which is causing the problem.

Moving from the pre-compiled FFmpeg Windows build "2015-04-28 git-b410c69" to "2015-07-02 git-03b2b40" caused this for me (changing back fixes it).
The latest pre-compiled Ubuntu Linux version appears to have the same problem.

If I run " ffprobe -show_format abc.mp4" on the command line against the old Windows and new Windows version it looks same when I scan it by eye, version numbers aside.

This only affects videos in portrait and I've only tried it with uploads from one device.

Stack trace:

e:\Mercurial\nodejs\iflicks\node_modules\fluent-ffmpeg\lib\ffprobe.js:153
              var legacyTagKeys = Object.keys(target).filter(legacyTag);
                                         ^
TypeError: Object.keys called on non-object
    at Function.keys (native)
    at e:\Mercurial\nodejs\iflicks\node_modules\fluent-ffmpeg\lib\ffprobe.js:153:42
    at Array.forEach (native)
    at handleExit (e:\Mercurial\nodejs\iflicks\node_modules\fluent-ffmpeg\lib\ffprobe.js:152:46)
    at ChildProcess.<anonymous> (e:\Mercurial\nodejs\iflicks\node_modules\fluent-ffmpeg\lib\ffprobe.js:191:11)
    at ChildProcess.emit (events.js:110:17)
    at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
6 Jul 16:07:44 - [nodemon] app crashed - waiting for file changes before starting...

**Comment by nickwhiteley** [comment]
A bit more info:
The output from "ffprobe -show_streams filename.mp4" has a new section called [SIDE_DATA] nested inside the [STREAM] section.  The additional layer of nesting breaks the parseFfprobeOutput function.

**Comment by nickwhiteley** [comment]
The problem is with an empty line returned by ffprobe in a new section called [SIDE_DATA].  The empty line causes one of the while loops to drop out.  
Adding "lines = lines.filter(Boolean);" to line 15 of /lib/ffprobe.js removes blank entries and fixes it.

**Comment by jdp** [comment]
I'm seeing this issue too. The fix from @nickwhiteley works for me as well.

**Comment by njoyard** [comment]
Will be fixed by #412, which I'll merge ASAP.

**Comment by brunosiqueira** [comment]
@nickwhiteley 's fix worked for me to! Is there a planed release with it?

**Comment by ashokfernandez** [comment]
+1

**Comment by snapptop** [comment]
confirmed.. master is working for both portrait and landscape on Windows using same FFMPEG Version as my first comment

@nickwhiteley apologies for dropping the ball on error trace and thank you for following up and debugging through it
@njoyard great work on this lib.. it is fantastic and really simple to work with .. {"thank":"you"}

```
"dependencies": {
    "fluent-ffmpeg": "git://github.com/fluent-ffmpeg/node-fluent-ffmpeg.git#master"
  }
```

![screenshots_working_yey fw](https://cloud.githubusercontent.com/assets/4693878/10419381/0ef82718-7044-11e5-819a-46d0902fbea0.png)

**Comment by lalitlogical** [comment]
@njoyard, I want to used this library, but it has contains the bug in screenshots of portrait videos. Can you please publish the changes for same as you stated above?

**Comment by njoyard** [comment]
I dont have quite the time to do a full version publish yet, as I would like to review some pull requests before.
You can use the master branch in the meantime (put the git url in your package.json instead of a version number).

**Comment by kryo2k** [comment]
I'm still getting this error, i have
```
    "fluent-ffmpeg": "git://github.com/fluent-ffmpeg/node-fluent-ffmpeg.git#master",
```
